src/initialization/initializeTypeScript/writeMultiTypeScriptConfig.ts

```diff
@@ -11,61 +12,74 @@ export interface MultiTypeScriptConfigSettings {
 	testFiles?: string;
 }
 
-export const writeMultiTypeScriptConfig = async ({
-	fileName,
+export const writeMultiTypeScriptConfig = async (
+	settings: MultiTypeScriptConfigSettings,
+) => {
+	const config = generateMultiTypeScriptConfig(settings);
+	await fs.writeFile(settings.fileName, JSON.stringify(config, undefined, 4));
+};
+
+export const generateMultiTypeScriptConfig = ({
 	improvements,
 	project,
 	sourceFiles,
 	testFiles,
 }: MultiTypeScriptConfigSettings) => {
-	await fs.writeFile(
-		fileName,
-		JSON.stringify(
-			[
-				{
-					fixes: {
-						...printImprovements(improvements),
-						strictNonNullAssertions: true,
-					},
-					...(testFiles && { include: [testFiles] }),
-					projectPath: project.filePath,
-					types: {
-						strictNullChecks: true,
-					},
-				},
-				{
-					...(testFiles && { exclude: [testFiles] }),
-					fixes: printImprovements(improvements),
-					...(sourceFiles && { include: [sourceFiles] }),
-					projectPath: project.filePath,
-				},
-				{
-					fixes: printImprovements(improvements),
-					...(testFiles
-						? { include: [testFiles, sourceFiles] }
-						: { include: [sourceFiles] }),
-					projectPath: project.filePath,
-				},
-			],
-			undefined,
-			4,
-		),
-	);
+	const fixes = printImprovements(improvements);
+
+	const stage1: Partial<RawTypeStatOptions> = {
+		fixes: {
+			...fixes,
+			strictNonNullAssertions: true,
+		},
+		projectPath: project.filePath,
+		types: {
+			strictNullChecks: true,
+		},
+	};
+
+	const stage2: Partial<RawTypeStatOptions> = {
+		fixes,
+		projectPath: project.filePath,
+	};
+
+	const stage3: Partial<RawTypeStatOptions> = {
+		fixes,
+		projectPath: project.filePath,
+	};
+
+	if (testFiles) {
+		// @ts-expect-error Cannot assign to 'include' because it is a read-only property.
```

Comment:
[Refactor] For a project that's supposed to migrate over to best TypeScript practices, having a bunch of `// @ts-expect-error`s in source code is probably not the best look 😅. A couple of options here:

* Using a type like [`Writable`](https://github.com/sindresorhus/type-fest/blob/cfb594777f5ede903c151a45c2db250b057902d7/source/writable.d.ts) for the three variables
* Having the variables declare the conditional properties: e.g. in `stage1`, add `include: testFiles ? [testFiles] : undefined`

I personally would go for the latter.